### PR TITLE
UIOAIPMH-21: add set spec specification on create/edit page

### PR DESCRIPTION
**Pull request by Dmitriy-Litvinenko**
# Purpose
Preview set specification on create/edit page
# Link
https://issues.folio.org/browse/UIOAIPMH-21
# Screenshot
image

### CODE REVIEW

File: src/settings/util/convertData.js [hunk 1]
```diff
@@ -52,7 +52,9 @@ export const filteringConditionsToFormData = (filteringConditions = [], setsFilt
 
 export const filteringConditionsToDtoFormat = (filteringConditions = []) => ({
   [SET_FIELDS.FILTERING_CONDITIONS]: filteringConditions
-    .filter(filteringCondition => filteringCondition[FILTERING_CONDITIONS_FIELDS.ACTIVE])
+    .filter((filteringCondition) => filteringCondition[FILTERING_CONDITIONS_FIELDS.ACTIVE])
+    .filter((filteringCondition) => filteringCondition[FILTERING_CONDITIONS_FIELDS.VALUE])
+    .filter((filteringCondition) => filteringCondition[FILTERING_CONDITIONS_FIELDS.SET_SPEC])
```
Comment (ViktorSoroka07) [inline src/settings/util/convertData.js:57]: I wonder whether three filter calls can be replaced with just one. Although if the dataset is small it is not a big deal.

File: src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import {
+  SET_FIELDS,
+  SET_FIELDS_INITIAL_VALUES,
+} from '../../../../../../../constants';
+
+import css from './SetSpecification.css';
+
+const generateSetSpecification = (value) => (
+  value && value.length
```
Comment (ViktorSoroka07) [inline src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js:16]: I suggest using isEmpty from lodash.
```suggestion
  isEmpty(value)
```

or `?.`

```suggestion
  value?.length
```

File: src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      >
+        {label}
+      </div>
+      <div
+        data-test-set-fields-set-specification-text
+        className={css.content}
+      >
+        {
+          generateSetSpecification(value)
+        }
```
Comment (ViktorSoroka07) [inline src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js:41]:
```suggestion
        {generateSetSpecification(value)}
```

File: src/settings/util/convertData.js (hunk 1 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/util/convertData.js]: Fixed

File: src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js (hunk 2 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js]: Fixed

File: src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js (hunk 3 above)
Comment (Dmitriy-Litvinenko) [reply src/settings/components/Sets/SetsForm/components/GeneralInformation/components/SetSpecification/SetSpecification.js]: Fixed

**Comment by Dmitriy-Litvinenko** [comment]
@ViktorSoroka07  Could you please re-review?

**Review by ViktorSoroka07 — APPROVED** [review]

**Comment by ViktorSoroka07** [comment]
@Dmitriy-Litvinenko, looks good.

**Review by ryandberger — APPROVED** [review]

**Pull request merged**